src/twisted/test/test_sslverify.py

```diff
@@ -368,33 +456,43 @@ def loopbackTLSConnectionInMemory(
     @type privateKey: L{str} (native string)
 
     @param serverCertificate: The certificate used by the server.
+
     @type chainedCertFile: L{str} (native string)
 
     @param clientProtocols: The protocols the client is willing to negotiate
-        using NPN/ALPN.
+        using ALPN.
 
     @param serverProtocols: The protocols the server is willing to negotiate
-        using NPN/ALPN.
+        using ALPN.
 
-    @param clientOptions: The type of C{OpenSSLCertificateOptions} class to
-        use for the client. Defaults to C{OpenSSLCertificateOptions}.
+    @param clientOptions: The type of C{OpenSSLCertificateOptions} class to use
+        for the client.  Defaults to C{OpenSSLCertificateOptions}.
 
     @return: 3-tuple of server-protocol, client-protocol, and L{IOPump}
     @rtype: L{tuple}
     """
     if clientOptions is None:
         clientOptions = sslverify.OpenSSLCertificateOptions
 
-    clientCertOpts = clientOptions(
-        trustRoot=trustRoot, acceptableProtocols=clientProtocols
-    )
-    serverCertOpts = sslverify.OpenSSLCertificateOptions(
-        privateKey=privateKey,
-        certificate=serverCertificate,
-        acceptableProtocols=serverProtocols,
-    )
-
-    return _loopbackTLSConnection(serverCertOpts, clientCertOpts)
+    if viaFactory:
+        clientCertOpts = clientOptions(trustRoot=trustRoot)
+        serverCertOpts = sslverify.OpenSSLCertificateOptions(
+            privateKey=privateKey, certificate=serverCertificate
+        )
+        return _loopbackTLSConnection(
+            serverCertOpts, clientCertOpts, serverProtocols, clientProtocols
+        )
+    else:
```

Comment:
Minor comment. Is this "else" needed ? Maybe just remove it as the other branch ends with a return